src/main/java/beam/analysis/plots/DeadHeadingStats.java

```diff
@@ -38,79 +47,157 @@ public void processStats(Event event) {
-        double count = 0;
-        for (double hourData : hoursData) {
-            count = count + hourData;
+    // Deadheading Distance Graph
+
+    private void processDeadHeadingDistanceRemainingRepositionings() {
+
+        Set<String> vehicleIds = vehicleEvents.keySet();
+
+        for (String vid : vehicleIds) {
+
+            Map<Integer, List<Event>> vehicleData = vehicleEvents.get(vid);
+
+            if (vehicleData != null) {
+                List<Integer> hourKeys = new ArrayList<>(vehicleData.keySet());
+                Collections.sort(hourKeys);
+
+                int n = hourKeys.size();
+                for (int k = 0; k < n; k++) {
+
+                    int hourKey = hourKeys.get(k);
+                    List<Event> vehicleHourData = vehicleData.get(hourKey);
+
+                    int m = vehicleHourData.size();
+                    if (k == (n - 1)) {
+                        m = vehicleHourData.size() - 1;
+                    }
+
+                    for (int i = 0; i < m; i++) {
+
+                        Event oldEvent = vehicleHourData.get(i);
+                        Double length2 = Double.parseDouble(oldEvent.getAttributes().get(PathTraversalEvent.ATTRIBUTE_LENGTH));
+
+                        updateDeadHeadingTNCMap(length2, hourKey, -1);
+                    }
+                }
+            }
         }
-        return (int) Math.ceil(count);
-    }
 
-    public int getPassengerPerTripCountForSpecificHour(int bucketIndex, String mode, int hour) {
-        double[][] dataset = buildDeadHeadingDataset(this.deadHeadingsMap.get(mode), mode);
-        double[] hoursData = dataset[bucketIndex];
-        return (int) Math.ceil(hoursData[hour]);
+        vehicleEvents.clear();
     }
-    private void processDeadHeading(Event event) {
+
+    private void processEventForTncDeadheadingDistanceGraph(Event event) {
+
         int hour = GraphsStatsAgentSimEventsListener.getEventHour(event.getTime());
-        Map<String, String> eventAttributes = event.getAttributes();
-        String mode = eventAttributes.get(PathTraversalEvent.ATTRIBUTE_MODE);
-        String vehicle_id = eventAttributes.get(PathTraversalEvent.ATTRIBUTE_VEHICLE_ID);
-        String graphName = getGraphNameAgainstModeAndVehicleId(mode,vehicle_id);
-        Integer _num_passengers = getPathTraversalEventNumOfPassengers(event, eventAttributes);
-        boolean validCase = isValidCase(graphName, _num_passengers);
-        if (validCase) {
-            updateNumPassengerInDeadHeadingsMap(hour,graphName,_num_passengers);
-        }
+        Map<String, String> attributes = event.getAttributes();
+        String mode = attributes.get(PathTraversalEvent.ATTRIBUTE_MODE);
+        String vehicle_id = attributes.get(PathTraversalEvent.ATTRIBUTE_VEHICLE_ID);
+        String graphName = getGraphNameAgainstModeAndVehicleId(mode, vehicle_id);
+        Integer _num_passengers = getPathTraversalEventNumOfPassengers(event);
```

Comment:
⚠️ This variable does not follow Java naming conventions. Consider using `numPassengers`